CRM/Contribute/BAO/Contribution.php

```diff
@@ -249,8 +249,8 @@ public static function add(&$params) {
 
     // Check whether this is linked to a recurring contribution and whether this is the first contribution.
     // If so we should create the template contribution for the recurring contribution.
-    if (self::isFirstContributionOfRecurringContribution($params, $contribution->id)) {
-      CRM_Contribute_BAO_ContributionRecur::createTemplateContributionFromFirstContribution($params['contribution_recur_id']);
+    if (self::recurringContributionRequiresTemplate($params, $contribution->id)) {
```

Comment:
I am not so sure about that. If we remove the if statement, it will check the recurring for every contribution. Even the contributions which do not belong to a recurring series.

Reply:
@artfulrobot We assumed that you did not need to have a template contribution because if one did not exist you could just use the latest contribution as the template. But I think @jaapjansma has commented on dev/financial#6 that he may have a use-case for there always being a template contribution.

Reply:
Yes this PR does create a template contribution for every recurring contribution.